presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java

```diff
@@ -220,4 +221,20 @@ public static Slice fromHexVarbinary(@SqlType(StandardTypes.VARBINARY) Slice sli
     {
         return fromHexVarchar(slice);
     }
+
+    @Description("compute crc32 hash")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice crc32(@SqlType(StandardTypes.VARBINARY) Slice slice)
+    {
+        Slice hash = Slices.allocate(Long.BYTES);
+        CRC32 crc32 = new CRC32();
+        crc32.update(slice.toByteBuffer());
+
+        // The expectation is that CRC32 is unsigned, so this won't fit
+        // correctly in a 32 signed integer, therefore this uses a long.
+        long crc32Value = crc32.getValue();
+        hash.setLong(0, Long.reverseBytes(crc32Value));
+        return hash;
```

Comment:
The function has been updated to return a long, following the comments in the pull request.